js-src/Builder.ts

```diff
@@ -77,47 +89,47 @@ export class Builder implements neon.Builder {
     );
   }
 
-  async toArchive(asset: neon.DestinationAsset): Promise<void> {
+  async toArchive(asset: DestinationAsset): Promise<void> {
     return neon.builderToArchive.call(this.builder, asset);
   }
 
-  static async fromArchive(asset: neon.SourceAsset): Promise<Builder> {
+  static async fromArchive(asset: SourceAsset): Promise<Builder> {
     return new Builder(await neon.builderFromArchive(asset));
   }
 
   sign(
-    signer: LocalSigner,
-    input: neon.SourceAsset,
-    output: neon.DestinationAsset,
+    signer: LocalSignerInterface,
+    input: SourceAsset,
+    output: DestinationAsset,
   ): Buffer {
     return neon.builderSign.call(this.builder, signer.signer(), input, output);
   }
 
   signFile(
-    signer: LocalSigner,
+    signer: LocalSignerInterface,
     filePath: string,
-    output: neon.DestinationAsset,
+    output: DestinationAsset,
   ): Buffer {
-    const input: neon.FileAsset = { path: filePath };
+    const input: FileAsset = { path: filePath };
     return neon.builderSign.call(this.builder, signer.signer(), input, output);
   }
 
   async signConfigAsync(
     callback: (data: Buffer) => Promise<Buffer>,
-    signerConfig: neon.JsCallbackSignerConfig,
-    input: neon.SourceAsset,
-    output: neon.DestinationAsset,
+    signerConfig: JsCallbackSignerConfig,
+    input: SourceAsset,
+    output: DestinationAsset,
   ): Promise<Buffer> {
     return neon.builderSignConfigAsync
       .call(this.builder, callback, signerConfig, input, output)
       .then((result) => {
         // output is a buffer and result is the manifest and the signed asset.
-        if ('buffer' in output) {
-          if ('signedAsset' in result && 'manifest' in result) {
+        if ("buffer" in output) {
```

Comment:
For a later time - if output is directly on buffer, you can check buffer.output.
The "in" check is only useful if it comes from a place in the inheritance/prototype chain and you're not sure it was defined directly on the output.

Reply:
Yes I think `in` is required. `DestinationAsset` is a union type.
`DestinationBufferAsset` has a buffer property and `FileAsset` does not.